cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -46,7 +46,7 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
   endif()
 
   find_package(Python2 COMPONENTS Interpreter Development NumPy)
-  if(Python2_Development_FOUND)
+  if(Python2_Development_FOUND AND ${Python2_VERSION} VERSION_GREATER_EQUAL "2.7")
```

Comment:
I think we should be a bit more informative, just adding this condition means that the user could see that CMake found a e.g. Python2.6 installation, and if there is no Python3, an error will appear saying "No Python 2 or 3 were found", which would be confusing since they saw we found their Python2.6. Or they expect that we will build with both Python2.6 and Python3, and in the end they only see the libraries generated for Python3 with no more information.

I believe we should first give a message here saying something like "Ignoring Python2 installation: unsupported version X.Y (version>=2.7 required)". And later, if no suitable Python2 nor Python3 is found, say `message(FATAL_ERROR "No supported Python 2 or 3 were found")`